data-prepper-plugins/csv-processor/src/main/java/com/amazon/dataprepper/plugins/processor/csv/CSVProcessor.java

```diff
@@ -45,4 +111,44 @@ public boolean isReadyForShutdown() {
     public void shutdown() {
 
     }
+
+    private CsvMapper createCsvMapper() {
+        CsvMapper mapper = new CsvMapper();
+        mapper.enable(CsvParser.Feature.WRAP_AS_ARRAY); // allows mapper to read with empty schema
+        return mapper;
+    }
+
+    private CsvSchema createCsvSchema() {
+        char delimiter_as_char = this.delimiter.charAt(0); // safe due to config input validations
```

Comment:
Could delimiter and quote be parsed as `char`s instead of Strings?

Reply:
Good callout! I had these as `char`s in the config originally. Asif and I talked about the config offline and decided to not use any primitive types. I'm not exactly sure but I believe using primitives in the config can cause weird behavior around null values. I think it was a bigger problem for `boolean` vs `java.lang.Boolean`. Do you forsee any problems with the String conversion implementation?

Reply:
The point about null conversion makes sense. I think the String conversion approach should be fine as is.